tests/network/hotplug_sriov.go

```diff
@@ -0,0 +1,150 @@
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+var _ = SIGDescribe("[Serial] SRIOV nic-hotplug", Serial, decorators.SRIOV, func() {
+
+	sriovResourceName := readSRIOVResourceName()
+
+	BeforeEach(func() {
+		Expect(checks.HasFeature(virtconfig.HotplugNetworkIfacesGate)).To(BeTrue())
+	})
+
+	BeforeEach(func() {
+		// Check if the hardware supports SRIOV
+		if err := validateSRIOVSetup(kubevirt.Client(), sriovResourceName, 1); err != nil {
+			Skip("Sriov is not enabled in this environment. Skip these tests using - export FUNC_TEST_ARGS='--skip=SRIOV'")
+		}
+	})
+
+	Context("a running VM", func() {
+		var hotPluggedVM *v1.VirtualMachine
+		var hotPluggedVMI *v1.VirtualMachineInstance
+
+		BeforeEach(func() {
+			By("Creating a VM")
+			hotPluggedVM = newVMWithOneInterface()
+			var err error
+			hotPluggedVM, err = kubevirt.Client().VirtualMachine(testsuite.GetTestNamespace(nil)).Create(context.Background(), hotPluggedVM)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(func() error {
+				var err error
+				hotPluggedVMI, err = kubevirt.Client().VirtualMachineInstance(testsuite.GetTestNamespace(nil)).Get(context.Background(), hotPluggedVM.GetName(), &metav1.GetOptions{})
+				return err
+			}, 120*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+			libwait.WaitUntilVMIReady(hotPluggedVMI, console.LoginToAlpine)
+
+			By("Creating a NAD")
+			Expect(createSRIOVNetworkAttachmentDefinition(testsuite.GetTestNamespace(nil), nadName, sriovResourceName)).To(Succeed())
+
+			By("Hotplugging an interface to the VM")
+			Expect(addSRIOVInterface(hotPluggedVM, ifaceName, nadName)).To(Succeed())
+		})
+
+		It("can be hotplugged a network interface", func() {
+			waitForSingleHotPlugIfaceOnVMISpec(hotPluggedVMI)
+			hotPluggedVMI = verifySriovDynamicInterfaceChange(hotPluggedVMI, migrationBased)
+			Expect(libnet.InterfaceExists(hotPluggedVMI, vmIfaceName)).To(Succeed())
+
+			updatedVM, err := kubevirt.Client().VirtualMachine(hotPluggedVM.Namespace).Get(context.Background(), hotPluggedVM.Name, &metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			vmIfaceSpec := vmispec.LookupInterfaceByName(updatedVM.Spec.Template.Spec.Domain.Devices.Interfaces, ifaceName)
+			Expect(vmIfaceSpec).NotTo(BeNil(), "VM spec should contain the new interface")
+			Expect(vmIfaceSpec.MacAddress).NotTo(BeEmpty(), "VM iface spec should have MAC address")
+
+			Eventually(func(g Gomega) {
+				updatedVMI, err := kubevirt.Client().VirtualMachineInstance(hotPluggedVMI.Namespace).Get(context.Background(), hotPluggedVMI.Name, &metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				vmiIfaceStatus := vmispec.LookupInterfaceStatusByName(updatedVMI.Status.Interfaces, ifaceName)
+				g.Expect(vmiIfaceStatus).NotTo(BeNil(), "VMI status should report the hotplugged interface")
+				g.Expect(vmiIfaceStatus.MAC).NotTo(BeEmpty(), "VMI hotplugged iface status should report MAC address")
+
+				g.Expect(vmiIfaceStatus.MAC).To(Equal(vmIfaceSpec.MacAddress),
```

Comment:
I think this should be out of the retry-loop. If there is a different mac address, it should just fall and not drag this for 30sec.

Reply:
Most of the comments on this file should've been raised when the original gotlpug e2e tests were introduced/reviewed. This file is just reusing/copying the same. 

Regarding the specific comment, it may take time for the MAc to be reported on the status, that's why it is inside the eventually.